kolibri/plugins/learn/assets/src/views/LearnIndex.vue

```diff
@@ -132,6 +132,15 @@
             };
           }
         }
+        if (this.pageName === PageNames.SEARCH) {
+          return {
+            appBarTitle: this.$tr('learnTitle'),
+            immersivePage: true,
+            immersivePageRoute: this.$router.getRoute(PageNames.TOPICS_ROOT),
```

Comment:
We should return the user to whichever page they were on when they initiated the search, since the search bar appears every where in the Learn page.

In the Coach > Lessons > Manage Resources workflow, we use query parameters in that flow's search URL to determine where to return the user when they press the "Exit search" button. We also use similar tricks in Coach reports to send users back to the right page when they view other immersive pages (like the exercise and exam report pages).

Reply:
@jonboiser I was thinking the same thing. I think that would make more sense and with your notes I imagine I'll have a fairly easy time getting that functionality.

@khangmach what are your thoughts?

Reply:
Returning the user to where they were previously is a good strat! 👍